### PR TITLE
[pytorch][1.10][training] fix typo in mpirun path

### PR DESCRIPTION
**Pull request by jeet4320**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
On EKS pod, if PATH is not updated with openmpi path `/opt/amazon/openmpi/bin`, then it throws below error:
`/opt/amazon/openmpi/bin/mpirun: line 2: mpirun.real: command not found`

To fix this, `mpirun.real` should have full path in DLC like `/opt/amazon/openmpi/bin/mpirun.real`

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by arjkesh** [comment]
Update this skip condition: https://github.com/aws/deep-learning-containers/blob/master/test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl.py#L61

There is no DGL in pt 1.10, it seems

File: test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def test_pytorch_train_dgl_gpu(pytorch_training, ec2_connection, gpu_only, py3_o
     _, image_framework_version = get_framework_and_version_from_tag(pytorch_training)
     image_cuda_version = get_cuda_version_from_tag(pytorch_training)
     # TODO: Remove when DGL gpu test on ec2 get fixed
-    if Version(image_framework_version) == Version("1.10") and image_cuda_version == "cu113":
+    if Version("1.10") <= Version(image_framework_version) < Version("1.11") and image_cuda_version == "cu113":
```
Comment (saimidu) [inline test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py:89]: Please change this to `Version(image_framework_version) in SpecifierSet("==1.10.*")`
```suggestion
    if Version(image_framework_version) in SpecifierSet("==1.10.*") and image_cuda_version == "cu113":
```

File: test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ def test_pytorch_train_dgl_gpu(pytorch_training, ec2_connection, gpu_only, py3_o
 def test_pytorch_train_dgl_cpu(pytorch_training, ec2_connection, cpu_only, py3_only):
     _, image_framework_version = get_framework_and_version_from_tag(pytorch_training)
     # TODO: Remove when DGL gpu test on ecs get fixed
-    if Version(image_framework_version) == Version("1.10"):
+    if Version("1.10") <= Version(image_framework_version) < Version("1.11"):
```
Comment (saimidu) [inline test/dlc_tests/ec2/pytorch/training/test_pytorch_training.py:102]: Please change this to `Version(image_framework_version) in SpecifierSet("==1.10.*")`
```suggestion
    if Version(image_framework_version) in SpecifierSet("==1.10.*"):
```

File: test/dlc_tests/ecs/pytorch/training/test_ecs_pytorch_training.py [hunk 3]
```diff
@@ -132,7 +132,7 @@ def test_ecs_pytorch_training_dgl_cpu(
     """
     _, image_framework_version = get_framework_and_version_from_tag(pytorch_training)
     # TODO: Remove when DGL gpu test on ecs get fixed
-    if Version(image_framework_version) == Version("1.10"):
+    if Version("1.10") <= Version(image_framework_version) < Version("1.11"):
```
Comment (saimidu) [inline test/dlc_tests/ecs/pytorch/training/test_ecs_pytorch_training.py:135]: Please change this to `Version(image_framework_version) in SpecifierSet("==1.10.*")`
```suggestion
    if Version(image_framework_version) in SpecifierSet("==1.10.*"):
```

File: test/dlc_tests/ecs/pytorch/training/test_ecs_pytorch_training.py [hunk 4]
```diff
@@ -163,7 +163,7 @@ def test_ecs_pytorch_training_dgl_gpu(
     if Version(image_framework_version) == Version("1.6") and image_cuda_version == "cu110":
         pytest.skip("DGL does not suport CUDA 11 for PyTorch 1.6")
     # TODO: Remove when DGL gpu test on ecs get fixed
-    if Version(image_framework_version) == Version("1.10") and image_cuda_version == "cu113":
+    if Version("1.10") <= Version(image_framework_version) < Version("1.11") and image_cuda_version == "cu113":
```
Comment (saimidu) [inline test/dlc_tests/ecs/pytorch/training/test_ecs_pytorch_training.py:166]: Please change this to `Version(image_framework_version) in SpecifierSet("==1.10.*")`
```suggestion
    if Version(image_framework_version) in SpecifierSet("==1.10.*") and image_cuda_version == "cu113":
```

File: test/dlc_tests/eks/pytorch/training/test_eks_pytorch_training.py [hunk 5]
```diff
@@ -172,7 +172,7 @@ def test_eks_pytorch_dgl_single_node_training(pytorch_training, py3_only):
     if Version(image_framework_version) == Version("1.6") and image_cuda_version == "cu110":
         pytest.skip("DGL does not suport CUDA 11 for PyTorch 1.6")
     # TODO: Remove when DGL gpu test on ecs get fixed
-    if Version(image_framework_version) == Version("1.10"):
+    if Version("1.10") <= Version(image_framework_version) < Version("1.11"):
```
Comment (saimidu) [inline test/dlc_tests/eks/pytorch/training/test_eks_pytorch_training.py:175]: Please change this to `Version(image_framework_version) in SpecifierSet("==1.10.*")`
```suggestion
    if Version(image_framework_version) in SpecifierSet("==1.10.*"):
```

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl.py [hunk 6]
```diff
@@ -38,7 +38,7 @@
 def test_dgl_gcn_training_cpu(ecr_image, sagemaker_regions, instance_type):
     # TODO: Remove when DGL gpu test on ecs get fixed
     _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
-    if Version(image_framework_version) == Version("1.10"):
+    if Version("1.10") <= Version(image_framework_version) < Version("1.11"):
```
Comment (saimidu) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl.py:41]: Please change this to `Version(image_framework_version) in SpecifierSet("==1.10.*")`
```suggestion
    if Version(image_framework_version) in SpecifierSet("==1.10.*"):
```

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl.py [hunk 7]
```diff
@@ -58,7 +58,7 @@ def test_dgl_gcn_training_gpu(ecr_image, sagemaker_regions, instance_type):
     image_cuda_version = get_cuda_version_from_tag(ecr_image)
     
     # TODO: Remove when DGL gpu test on ecs get fixed
-    if Version(image_framework_version) == Version("1.10") and image_cuda_version == "cu113":
+    if Version("1.10") <= Version(image_framework_version) < Version("1.11") and image_cuda_version == "cu113":
```
Comment (saimidu) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_dgl.py:61]: Please change this to `Version(image_framework_version) in SpecifierSet("==1.10.*")`
```suggestion
    if Version(image_framework_version) in SpecifierSet("==1.10.*") and image_cuda_version == "cu113":
```

File: dlc_developer_config.toml [hunk 8]
```diff
@@ -20,21 +20,21 @@ benchmark_mode = false
...
-sanity_tests = true
+sanity_tests = false
   safety_check_test = false
   ecr_scan_allowlist_feature = false
-ecs_tests = true
-eks_tests = true
-ec2_tests = true
+ecs_tests = false
+eks_tests = false
+ec2_tests = false
```
Comment (saimidu) [inline dlc_developer_config.toml:37]: Please run the sanity/ecs/eks/ec2 tests, since the DGL tests have had their skip-logic modified.

**Review by saimidu — COMMENTED** [review]
Conditionally approved, if the EC2/ECS/EKS tests all pass without issue.

**Review by saimidu — DISMISSED** [review]
Approved. Please revert temp changes.

**Review by saimidu — APPROVED** [review]

**Pull request merged**